include/ignition/gazebo/components/Serialization.hh

```diff
@@ -183,6 +183,40 @@ namespace serializers
       return _in;
     }
   };
+
+  template <typename T>
+  class VectorSerializer
+  {
+    /// \brief Serialization for `std::vector<T>` with serializable T.
+    /// \param[in] _out Output stream.
+    /// \param[in] _data The data to stream.
+    /// \return The stream.
+    public: static std::ostream &Serialize(std::ostream &_out,
+      const std::vector<T> &_data)
+    {
+      _out << _data.size();
```

Comment:
I think we need to add a whitespace charactor to `_out` after adding `_data.size()` and after adding each `datum`. Currently, if your vector is `[{5,6},{7,8}]`, I think it'll end up being serialized as `25 67 8`.

Reply:
I added the space. I hope the deserialization step doesn't need any change.